src/condor_daemon_core.V6/daemon_core.cpp

```diff
@@ -8916,12 +8916,60 @@ int extractInheritedSocks (
 	return cSocks;
 }
 
+#ifdef WIN32
+static char * redact_condor_inherit(char* redacted_env)
+{
+	if (redacted_env) return redacted_env;
+
+	// save CONDOR_INHERIT and CONDOR_PRIVATE_INHERIT into a malloc'ed buffer formatted as environment
+	std::string redacted, inherit, private_inherit;
+
+	GetEnv(ENV_CONDOR_INHERIT, inherit);
+	GetEnv(ENV_CONDOR_PRIVATE, private_inherit);
+
+	if ( ! inherit.empty()) {
+		redacted += ENV_CONDOR_INHERIT;
+		redacted += "=";
+		redacted += inherit;
+		redacted.push_back(0);
+	}
+	if ( ! private_inherit.empty()) {
+		redacted += ENV_CONDOR_PRIVATE;
+		redacted += "=";
+		redacted += private_inherit;
+		redacted.push_back(0);
+	}
+	if (redacted.empty()) {
+		return redacted_env;
+	}
+
+	redacted.push_back(0); // terminating null
+
+	// copy redacted string to a permanent mallocated buffer
+	int cb = (int)redacted.size();
+	redacted_env = (char*)malloc(cb);
+	memset(redacted_env, 0, cb);
+	memcpy(redacted_env, redacted.data(), redacted.size());
+
+	// publish the pid,size, and address of the redacted buffer in our environment
+	// so that programs that can read our process memory can retrieve the redacted environment
+	redacted.clear();
+	formatstr(redacted, "CONDOR_DCADDR=%d,%d,%lld", getpid(), (int)cb, redacted_env);
```

Comment:
Normally, I'd say use '%p' to format a memory address. Since the value is not being parsed by a scanf() or meant to be read by a human, the decimal format is probably fine. But you should at least make it unsigned.